### ISSUE TITLE
DHCP connections

### ISSUE THREAD

**Issue by joeman2116** [ManiacalLabs/ESPSerialWiFiManager#14]
Great code.  I integrated it with FAUXMOESP.

The issue I have is, I cannot connect  unless I do a static advanced setup - then commit.
Once that is done and I reboot - it connects very well.  I can change the ip , gateway and it still works fine.
However, if i scan and select the SSID then enter the password - it times out...

Any ides what might be the cause?

Thanks

Joe

**Comment by adammhaile** [comment]
Not really...would have to see logs at least. Most likely is a problem with the access point.

**Comment by joeman2116** [comment]
Ok,  
I have several other esp8266 devices and using other code they connect and get an ip with no problems.
I tried a few different esp devices and they behaved the same. 
I can see it attempting to connect  but eventually times out..

The codes seem to work well with each other.  The Code provides 2 Alexa devices by emulating a wemo switch...
The code below is what I am using:
Thanks for having a look...
Joe
-----------------------------------------------code-------------------------------------
#include <Arduino.h>
#include <ESP8266WiFi.h>
#include <ESPSerialWiFiManager.h>

#include "fauxmoESP.h"

#define SERIAL_BAUDRATE                 115200

fauxmoESP fauxmo;

#define Flex2_PIN 13 // d7=gpio13
#define Flex1_PIN 5 //d1 =gpio5



void callback(uint8_t device_id, const char * device_name, bool state) {
  Serial.printf("[MAIN] %s state: %s\n", device_name, state ? "ON" : "OFF");

  if ( (strcmp(device_name, "Flex1") == 0) ) {

    if (state) {
      digitalWrite(Flex1_PIN, HIGH);
    } else {
      digitalWrite(Flex1_PIN, LOW);
    }
  }

  if ( (strcmp(device_name, "Flex2") == 0) ) {
    // adjust the Flex immediately!
    if (state) {
      digitalWrite(Flex2_PIN, HIGH);
    } else {
      digitalWrite(Flex2_PIN, LOW);
    }
  }
}

void setup() {

  pinMode(Flex2_PIN, OUTPUT);
  digitalWrite(Flex2_PIN, LOW);//

  pinMode(Flex1_PIN, OUTPUT);
  digitalWrite(Flex1_PIN, LOW);

  //Create the manager object
  ESPSerialWiFiManager esp = ESPSerialWiFiManager();

  //void setup(){
  //Initialize the serial connection. This is required.
  Serial.begin(115200);
  //Complete manager init (this will init WiFi and EEPROM)
  esp.begin();
  //If desired at beginnging, show manager menu
  //This will timeout after 10 seconds
  esp.run_menu(10);


  // Fauxmo
  fauxmo.addDevice("Flex2");
  fauxmo.addDevice("Flex1");
  fauxmo.onMessage(callback);
}

void loop() {
  fauxmo.handle();

}

**Comment by adammhaile** [comment]
What's the IP of your DHCP host?

**Comment by joeman2116** [comment]
192.168.1.1 (router ip) dhcp host

**Comment by adammhaile** [comment]
Send me the serial log of when you use advanced mode and it does work.
Please remember to blank out the wifi password :)

**Comment by joeman2116** [comment]
### .....
Connection Sucessful
=============================
Current Network Details:
=============================
SSID: ASUS66ujoe
IP Address: 192.168.1.5
MAC address: 33:6C:29:7F:CF:5C
Subnet Mask: 255.255.255.0
Gateway: 192.168.1.1
DNS 1: 8.8.8.8
DNS 2: 192.168.1.1
=============================
Did you mean the above?  - from my serial monitor

**Comment by adammhaile** [comment]
I meant ALL of it. Not just the final details.

**Comment by joeman2116** [comment]
Sorry, here is the complete screen
![screen shot 03-07-17 at 11 08 pm](https://cloud.githubusercontent.com/assets/23128419/23688462/2019f432-038b-11e7-8e60-c7df44433aac.PNG)

**Comment by joeman2116** [comment]
I just tried to connect to another router and got the same results.. Only connects with a static ip advanced configuration

joe

**Comment by adammhaile** [comment]
Very weird... What version of the ESP arduino core are you using and what
actual board are you using?

**Comment by joeman2116** [comment]
board is using an esp8266 12 d1 mini
Not sure where to find info on core?
arduino ide is v 1.81
I have several other nodemcu - i could program - ?

**Comment by adammhaile** [comment]
Honestly, I'm not sure. It should work with DHCP. Go into the arduino
boards manager and search for ESP8266 and tell me what version it says
there.

**Comment by joeman2116** [comment]
The board manager says:
esp8266- builtin by Simon peter is ver 1.0.0

**Comment by adammhaile** [comment]
Current official release is 2.3.0: https://github.com/esp8266/Arduino
That is the only version my library supports.

**Comment by joeman2116** [comment]
Just as a test I loaded a new arduino ide v 1.81 with the latest core etc
Compiled - and upload fine

hit the reset and menu came up on serial monitor
selected 1 to scan and it found my wifi ssid fine
Entered password
said connecting to my wifi name
a bunch of dots then said Failed connecting to AP.

Did a static and advanced - then a commit----- logged in perfect..


joe

**Comment by joeman2116** [comment]
Just setup a guest account - with no password and it connected fine and got
a dhcp ip ok.
Changed to encrypted wpa and wpa2 - failed log in.

How do i erase the eeprom - ?
When i do a static with a new ip and dns etc , then do a commit - it still
remembers the old settings

joe

**Comment by adammhaile** [comment]
Choose option 4 (disconnect) then option 5 (commit) and that will clear out
the settings. Must be something with your specific wifi access point. I
used WPA all the time without issue. What model is it?

**Comment by joeman2116** [comment]
Wooo - It worked

The solution seems to  be i needed to 4 disconnect and 5 commit.
It appears it was remembering the dhcp connect with a bad pw - so  select 4
and 5 cleared all...
Once i did that i did a scan , selected my wifi and entered the password --
logged in!!!!
Rebooted  and it got a dhcp ip --- fine..

*Thanks for your help.*....


One more question:

I use fauxmoesp to select Alexa (2)devices called flex and flex1... with
your code and it works great....

I was wondering if you might know how to have the alexa device names
selectable on a menu ?
It would be great if one could just use a menu like your wifi code to
select  new names.

The idea is to be able to give a esp module to someone and use it on teir
wifi- which your code allows.
But also so they could change the name of the alexa device name device to
something more meaning full to them

joe

**Comment by adammhaile** [comment]
Right now, no... though I've had requests for opening up that menu to
custom things.
I think what I may do in the future is pull all the menu stuff out into a
separate class that can be passed to the WiFi manager but then you could
continue to build your own menus with it.
